### ISSUE TITLE
Local URL for media instead of Cloudinary URL in rest api

### ISSUE THREAD

**Issue by cfab** [cloudinary/cloudinary_wordpress#670]
This is a big pbl when using Wordpress as a headless CMS...

**Comment by Vdeub-cloudinary** [comment]
@cfab could you share more details on this? Are you using a Custom REST API endpoint or the core one?

**Comment by cfab** [comment]
Hello, 
with the version <3, the given url for a media in the rest api (or using wp-graphql) was the cloudinary media url
Since version 3, the given url is the local url, why did you change this ?
see screenshot below (now and then before v3) - or is it just me ? 
<img width="1352" alt="Capture d’écran 2022-01-27 à 12 23 52" src="https://user-images.githubusercontent.com/46970127/151349833-277402e2-805c-43d9-8395-975dc2cb2348.png">
<img width="1349" alt="Capture d’écran 2022-01-27 à 12 24 34" src="https://user-images.githubusercontent.com/46970127/151350119-02073759-f213-43ca-9b5e-19ca4b747208.png">

**Comment by ashleynexvelsolutions** [comment]
Any update on this? @cfab @Vdeub-cloudinary

**Comment by cfab** [comment]
no answer !

**Comment by Vdeub-cloudinary** [comment]
@ashleynexvelsolutions @cfab - we are working on this and hope to have a fix for `3.0.2`.
I'll keep you posted on this.

**Comment by cfab** [comment]
ok thank you !

**Comment by qdz90** [comment]
@cfab is this fixed in 3.0.2?

**Comment by Vdeub-cloudinary** [comment]
@qdz90 - it is fixed when requesting the content but not when requesting image nodes. For the image nodes, it is fixed in `3.0.3` that will be released next week.

**Comment by Vdeub-cloudinary** [comment]
We have released `3.0.3` that fixes this issue. Please upgrade and let us know if you still encounter any issues. Thanks!

**Comment by thrugl** [comment]
After updating _to_ 3.0.3 from 2.x.x, I am now experiencing this problem

**Comment by aleksandar-cloudinary** [comment]
Hi @thrugl, Could you please share the query you're running and the output?

**Comment by Vdeub-cloudinary** [comment]
Hi @thrugl, to add to Aleksandar's message. Could you please open a ticket to support@cloudinary.com with a [system report](https://support.cloudinary.com/hc/en-us/articles/360022260599-How-to-generate-a-system-report-on-the-WordPress-plugin-)? If you can also share if it is happening for all images or specific ones that could also be helpful.

**Comment by LarsEjaas** [comment]
I am experiencing the same issue using the Cloudinary WordPress plugin v. 3.0.9 and WPGraphQL plugin v. 1.13.9. 
If I check the URL inside WordPress admin when editing Posts and Pages I get the correct Cloudinary media URL, But when using the GraphQL API endpoint (or using the GraphiQL explorer) I get the WordPress image url. 
I have tried downgrading the Cloudinary plugin, however v. 2.7.6 of the plugin crash in the post editor of WordPress v. 6.1.1

**Comment by Vdeub-cloudinary** [comment]
Hi @LarsEjaas, could you please open a ticket to support@cloudinary.com with a [system report](https://support.cloudinary.com/hc/en-us/articles/360022260599-How-to-generate-a-system-report-on-the-WordPress-plugin-) and an example of GraphQL Query you are doing? Thanks!

**Comment by LarsEjaas** [comment]
I am sorry for not getting back earlier!

After some debugging I have narrowed the issue I am experiencing to the use of an authorization plugin I use to set CORS headers to my headless frontend:
https://github.com/funkhaus/wp-graphql-cors

I guess this is probably out of the scope of the support from you guys 🙂 But this is how it can be replicated:

1. Install the [WPGrapQL WordPress plugin](https://wordpress.org/plugins/wp-graphql/), the [WP Graphql Cors plugin](https://github.com/funkhaus/wp-graphql-cors) and the Cloudinary WordPress Plugin. Make sure the default "Sample Page" is published and add a `featured image` to the page hosted on Cloudinary.

2. Rightclick the featured image and copy the image URL to make sure it points to a Cloudinary URL.

3. Go to the: `GraphiQL IDE` menu point inside WP admin.

4. Inside the IDE add the following graphQL query:
```
query GET_PAGE(
  $uri: ID!
) {
  page: page(id: $uri, idType: URI) {
    featuredImage {
      node {
        sourceUrl
      }
    }
  }
}
```
5. In the `QUERY VARIABLES` field just below the field for queries, add the following JSON object:
```
{"uri": "sample-page"}
```
6. Click the `Execute Query` ▶️ button inside the GraphiQL IDE.

## Expected behaviour
I expect the image slug/URL to point to Cloudinary. I got a slug/URL pointing to the WordPress domain.

**Comment by Vdeub-cloudinary** [comment]
@LarsEjaas 
Thanks for the detailed explanation. 
I will try to reproduce the issue on my side and see if there is anything we can do.
I'll keep you posted if I find anything interesting.

**Comment by LarsEjaas** [comment]
> @LarsEjaas Thanks for the detailed explanation. I will try to reproduce the issue on my side and see if there is anything we can do. I'll keep you posted if I find anything interesting.

Thanks @Vdeub-cloudinary - please reach out if you need further details!

**Comment by Vdeub-cloudinary** [comment]
@LarsEjaas we have released `3.1.6` that should fix this issue.
Please let us know if it is not the case!
